LPS-111202 Move algo parameters passing to previous builder step

csierra opened the pull request:
hey @arthurchan35, 
I believe this looks good... I made just a couple of changes, let me know your thoughts. 

As we discussed I believe we can proceed with the portal part. I have also been discussing with Preston about petra modules and it looks like it would be a good fit for that. 

Let's talk on Tuesday. 

Bests. 

Carlos.

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

csierra commented:
hey Arthur, 
I rebased the pull and added some more changes to the interface because it seems to save code and more clear to me. Let me know if you don't like it or if you find problems with the approach. 

I have also discussed turning the lib into a petra module. It needs to depend only on other petra libraries and also have 100% test coverage. I think it might make sense to do that for the modules that we can and have others contribute hash providers that are outside petra. But I believe we can discuss this later since it is not urgent. For the portal implementation it should not matter where the library is defined. 

Talk to you tomorrow!
